sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/CompletionListener.kt

```diff
@@ -0,0 +1,14 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+/**
+ * A listener that's notified when the action is completed.
+ */
+fun interface CompletionListener {
+  /**
+   * Notifies the action is completed.
+   *
+   * @param isCanceled true if the transition was canceled and false
+   * if it ran to completion.
+   */
+  fun onComplete(isCanceled: Boolean)
```

Comment:
I found a way to eliminate the isCanceled flag while adding some tests on iOS. It should simplify creating ViewportTransition implementations.

Reply:
@macdrevx I think it's still useful in the Viewport, as 
```
  fun transitionTo(targetState: ViewportState, completionListener: CompletionListener? = null)
```
how would we notify if the transition is canceled without this `isCanceled` flag?

I think it might make sense to remove the flag in the ViewportTransition, as we can simply not trigger the `ViewportState.startUpdatingCamera()`.

Reply:
WDYT about using two separate callbacks, `onComplete`/`onCancel` instead of single one with flag? 
IMHO such flag is pretty error-prone, requires to remember (or check the docs) each time use use it, whether it's `true = completed` or `true = canceled`

Reply:
From discussion: we'll split this interface into 2: one for the ViewportTransition.run method that will *not* contain a flag, and one for the Viewport.transition(to:completion:) method that will contain a flag named `isFinished`.

@pengdev what do you think of @yunikkk's idea?

Reply:
using 2 listeners make the API surface larger but make the logic clearer, I think current abstraction is more efficient and the documentation is clear enough to use, wydt?